src/uvwasi.c

```diff
@@ -1925,8 +1925,11 @@ uvwasi_errno_t uvwasi_path_readlink(uvwasi_t* uvwasi,
     return uvwasi__translate_uv_error(r);
   }
 
+#ifdef __EMSCRIPTEN__
+  len = strlen(req.ptr);
```

Comment:
Can you explain why this is necessary?

Reply:
well, `wasm3` depends on `uvwasi` and while compiling `wasm3` with emscripten's `emcc` I would get an error stating that `strnlen` is not defined and I fix the problem using `strlen` instead, the difference should be (not scaning beyond the first `buf_len` bytes of the `req.ptr` string)

Reply:
Could this be some sort of configuration error on your end? This file (`src/uvwasi.c`) includes `string.h`, and emcc seems to support `strnlen()`. The following code:

```c
#include <stdio.h>
#include <string.h>

int main(void) {
  char* foo = "the quick brown fox jumped over the lazy dog";
  int size = strnlen(foo, 100);
  printf("size is %d\n", size);
}
```

Compiles and runs for me with the following command:

```
docker run --rm -v `pwd`:/src trzeci/emscripten emcc /src/test.c
```

Reply:
Yeah true that it works on plain emcc example .. that must be a bad configuration from my side, closing pull request